encoding/wkb/wkb.go

```diff
@@ -178,6 +178,12 @@ func Read(r io.Reader, opts ...wkbcommon.WKBOption) (geom.T, error) {
 				return nil, err
 			}
 		}
+		// If Z and/or M is specified, mark the collection with a fixed layout.
+		if gc.Empty() && layout != geom.XY {
```

Comment:
(i only did these for empty ones because there may be people using something like `GEOMETRYCOLLECTIONZ (POINT(0 0))` in WKB form, not sure whether we wanted to break those)